packages/atlas-signin/src/main.ts

```diff
@@ -0,0 +1,142 @@
+import { ipcMain, shell } from 'electron';
+import { URLSearchParams } from 'url';
+import * as plugin from '@mongodb-js/oidc-plugin';
+import fetch from 'node-fetch';
+import type { IntrospectInfo, Token, UserInfo } from './shared';
+import { Events } from './shared';
+
+export class AtlasSignIn {
+  private constructor() {
+    // singleton
+  }
+
+  private static calledOnce = false;
+
+  private static plugin = plugin.createMongoDBOIDCPlugin({
+    // TODO: as auth is done through the localhost loopback and not directly
+    // through customer okta, we will need to implement screens for following
+    // states: initial, success, error, not found; waiting for the designs
+    // redirectServerRequestHandler() {
+    // }
```

Comment:
For this one, I’d suggest moving the `redirectServerRequestHandler` implementation that already exists in the `devtools-connect` package into its own package and then using that, I don’t think we really need different pages here right now.

(We may just want to move `devtools-connect` to the share monorepo, makes it even easier to do that split, how does that sound?)

Reply:
Sure, I can include this work in the project and tell design that we will not be implementing these new sign in flow states they apparently were working on. We will also need to modify the server not to serve a blank page on initial redirect and provide an additional handler for that

Reply:
We were chatting about this during design sync on Thursday, and yes, we brought up that it probably makes sense to focus on designing the interfaces inside Compass itself. IIUC at least the actual auth flow pages in the customer Okta are not something we can design anyway?

Reply:
I'm a bit lost now, sorry, so what was the result of the discussion: do we want the generic devtools connect screens for the login or special compass one that they seem to start working on? And yeah, we can't affect customer okta screens, but this I highlighted already some time ago in one of the figma discussions

Reply:
Right, we want the generic screens (that already we use for DB OIDC login) for the initial iteration here :+1:

Reply:
Got it, thanks for clarifying!